tasks/release/osx.js

```diff
@@ -78,21 +81,65 @@ var renameApp = function () {
     });
     // Rename application
     finalAppDir.rename('Contents/MacOS/Electron', manifest.productName);
-    return Q();
+    return new Q();
 };
 
 var signApp = function () {
-    var identity = utils.getSigningId();
-    if (identity) {
+    var identity = utils.getSigningId(manifest);
+    var MASIdentity = utils.getMASSigningId(manifest);
+    var MASInstallerIdentity = utils.getMASInstallerSigningId(manifest);
+
+    if (utils.releaseForMAS()) {
+        if (!MASIdentity || !MASInstallerIdentity) {
+            gulpUtil.log('--mas-sign and --mas-installer-sign are required to release for Mac App Store!');
+            process.exit(0);
+        }
+        var cmds = [
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libffmpeg.dylib"',
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/Electron Framework.framework/Versions/A/Libraries/libnode.dylib"',
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/Electron Framework.framework/Versions/A"',
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/' + manifest.productName + ' Helper.app/"',
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/' + manifest.productName + ' Helper EH.app/"',
+            'codesign --deep -f -s "' + MASIdentity + '" --entitlements child.plist -v "' + finalAppDir.path() + '/Contents/Frameworks/' + manifest.productName + ' Helper NP.app/"'
+        ];
+
+        if (finalAppDir.exists('Contents/Frameworks/Squirrel.framework/Versions/A')) {
+            // # Signing a non-MAS build.
```

Comment:
btw there is already abstracted npm module for signing app: https://www.npmjs.com/package/electron-osx-sign
can be used like https://github.com/ubergrape/grape-electron/blob/master/tasks/release/osx.js#L96